src/app/models/submission.server.model.ts

```diff
@@ -255,42 +256,82 @@ EncryptSubmissionSchema.statics.findSingleMetadata = function (
   formId: string,
   submissionId: string,
 ): Promise<StorageModeSubmissionMetadata | null> {
-  return (
-    this.findOne(
-      {
-        form: formId,
-        _id: submissionId,
+  const pageResults: Promise<MetadataAggregateResult[]> = this.aggregate([
+    {
```

Comment:
shall we make the aggregate query to a single variable, since it seems to be shared across findSingleMetadata and findAllMetadata?

Reply:
@mayying could I get some clarifications for this? Single metadata has `submissionId` but findAllMetadata doesn't.

Reply:
I think the aggregateQuery function can accept an optional param submissionId. This should be the only difference right? other than that the entire query is the same?

Reply:
8449d3f55